### ISSUE TITLE
dockerd not starts, if no kernel modules locally available for modprobe (if modprobe not works)

### ISSUE THREAD

**Issue by inkvizitor68sl** [moby/moby#38209]
https://github.com/moby/moby/blob/b3e9f7b13b0f0c414fa6253e1f17a86b2cff68b5/daemon/graphdriver/overlay/overlay.go#L204

Since you changed some logick, and now you just run "modprobe" - dockerd will not start, if overlay module (or kind of) loaded, but modprobe cannot find this module in files locally. 

In console it looks like: 
# lsmod | grep overl
overlay                45056  0
# modprobe overlay
modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/4.14.105/modules.dep.bin'
modprobe: FATAL: Module overlay not found in directory /lib/modules/4.14.105

This can be reproduced in situations like: 
1) modprobe forbidden or removed for security reasons (and always rets exit 1+).
2) dockerd runs inside lxc or other kind of chroot-based isolation without kernel packages installed inside (module loaded from base system). 

In docker-ce=17.09.0~ce-0~ubuntu dockerd started normally in this situation.

**Comment by thaJeztah** [comment]
The code you're linking to hasn't modified since 4 years, so I doubt anything changed there.

could it be that error you're running into is related to https://github.com/docker/for-linux/issues/475 ?

can you post the logs of the error you're seeing?

**Comment by inkvizitor68sl** [comment]
Hi, thanks for attention. Yes, looks like i pointed wrong part of code. 
I am not really expirienced much to navigate through moby repo, but if you can point how to get diff between docker-ce=18.06.1~ce~3-0~ubuntu and 5:18.09.0~3-0~ubuntu-xenial, i can try to look what part of changes can be near this problem (cant find tags/branch for them). 

Logs for 5:18.09.0~3-0~ubuntu-xenial (broken for me) looked so: 
```
Nov 15 21:01:45 hostname_here modprobe[33787]: modprobe: ERROR: ../libkmod/libkmod.c:586 kmod_search_moddep() could not open moddep file '/lib/modules/4.14.105/modules.dep.bin'
Nov 15 21:01:45 hostname_here modprobe[33787]: modprobe: FATAL: Module overlay not found in directory /lib/modules/4.14.105
```
overlay module loaded before starting dockerd. 

I have some warnings about kernel modules in 18.06.1~ce~3-0~ubuntu, but this modules preloaded too, dockerd starts correctly and does things, which requires this modules (e.g. warning for nf_nat, but all nat functions works).

**Comment by inkvizitor68sl** [comment]
https://github.com/containerd/containerd/pull/2776 
looks like this pull request will fix issue.

**Comment by thaJeztah** [comment]
Thanks! Let me close this issue as a duplicate of https://github.com/docker/for-linux/issues/475. The fix in containerd () is also back ported to the internal packaging repository used to build the docker (and containerd) packages, and will be included in a 18.09.1 patch release.

To resolve the issue for now, I wrote down steps to manually update the systemd service in this comment: https://github.com/docker/for-linux/issues/475#issuecomment-437373774

You can subscribe to the linked issue to stay informed on updates. I'll lock the conversation on this ticket to prevent the discussion from diverging (but feel free to participate in the discussion on the linked issue)
